compiler/noirc_frontend/src/monomorphization/tests.rs

```diff
@@ -0,0 +1,37 @@
+#![cfg(test)]
+use crate::tests::get_program;
+
+use super::{ast::Program, errors::MonomorphizationError, monomorphize};
+
+fn get_monomorphized(src: &str) -> Result<Program, MonomorphizationError> {
```

Comment:
In the top-level frontend tests there's `monomorphize_program` and `get_monomorphization_error` which could be reused here.

(we could also have another type of `check_errors` calls `check_errors_with_monomorphization` to use squiggles, but that could be done in a separate PR)

Reply:
I ended up removing the one in the frontend since it didn't assert that there were no errors.

I also thought about moving the arithmetic generics tests that called monomorphization functions to the monomorphization tests but thought it'd be better if we kept features together rather than splitting them. I suppose there's an argument for not having a separate monomorphization tests module at all then and just doing it by feature. I think it's fine as long as it is miscellaneous things / single tests though.